sos/collector/sosnode.py

```diff
@@ -367,7 +367,9 @@ def _regex_sos_help(self, regex, sosinfo, is_list=False):
             for line in result.splitlines():
                 if not is_list:
                     try:
-                        res.append(line.split()[0])
+                        ls = line.split()
+                        if ls:
```

Comment:
If we're concerned about empty lines, and we're concerned about performance given the earlier conversation, then we should be applying the conditional before the `.split()`.

```python
for line in results.splitlines():
    if not is_list:
        if not line:
            continue
    [...]
```

Reply:
I tried this, and it covers all the empty lines, but there's a case where we print a single space and the exception gets printed. Playing with the output, this is what we get:

```
'a python.hashes             off             collect hashes for all python files'
'a rpm.rpmq                  on              query package information with rpm -q'
'a rpm.rpmva                 off             verify all packages'
'a rpm.rpmdb                 off             collect /var/lib/rpm'
'a selinux.fixfiles          off             collect incorrect file context labels'
'a services.servicestatus    off             collect status of all running services'
'a ssh.userconfs             on              Changes whether module will collect user .ssh configs'
''
' '
[localhost:_regex_sos_help] Error parsing sos help: list index out of range
```

That space is coming from here:

https://github.com/sosreport/sos/blob/main/sos/report/__init__.py#L1093

Any simple way to cover this case as well, without having to change the code in report? 
A walrus operator here would be more useful and cover all cases, but we are still waiting for  a decision on #3533 , right?

Reply:
@jcastill walrus is good, we already discussed this :)

Reply:
Perfect, I'll push a new version soon